keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepository.kt

```diff
@@ -192,10 +193,9 @@ class CombinedRepository(
 
   private fun Set<DeliveryArtifact>.transform(deliveryConfigName: String) =
     map { artifact ->
-      when (artifact) {
-        is DockerArtifact -> artifact.copy(deliveryConfigName = deliveryConfigName)
-        is DebianArtifact -> artifact.copy(deliveryConfigName = deliveryConfigName)
-      }
+      val asMap: MutableMap<String, Any?> = objectMapper.convertValue(artifact)
+      asMap["deliveryConfigName"] = deliveryConfigName
+      objectMapper.convertValue(asMap, artifact.javaClass)
```

Comment:
This is a bit hacky, but at least we don't have to make assumptions about the types. 🤷‍♂️